pkg/component/component.go

```diff
@@ -373,33 +372,32 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 	targetPath := fmt.Sprintf("%s/src", s2iSrcPath)
 
 	// Copy the files to the pod
-
-	s := log.Spinner("Copying files to pod")
+	spin := log.Spinner(true, "Copying files to pod")
 	err = client.CopyFile(path, pod.Name, targetPath, files)
 	if err != nil {
-		s.End(false)
+		spin.End(false)
 		return errors.Wrap(err, "unable push files to pod")
 	}
-	s.End(true)
+	spin.End(true)
 
-	s = log.Spinner("Building component")
+	// Create a spinner for the component being built
+	var s *log.Status
+	if show {
+		s = log.Spinnerf(false, "Building component")
+	} else {
+		s = log.Spinnerf(true, "Building component")
+	}
 	defer s.End(false)
 
-	// use pipes to write output from ExecCMDInContainer in yellow  to 'out' io.Writer
 	pipeReader, pipeWriter := io.Pipe()
 	var cmdOutput string
 	go func() {
-		yellowFprintln := color.New(color.FgYellow).FprintlnFunc()
 		scanner := bufio.NewScanner(pipeReader)
 		for scanner.Scan() {
 			line := scanner.Text()
-			// color.Output is temporarily used as there is a error when passing in color.Output from cmd/create.go and casting to io.writer in windows
-			// TODO: Fix this in the future, more upstream in the code at cmd/create.go rather than within this function.
-			// If we are in debug mode, we should show the output
-			if log.IsDebug() {
-				yellowFprintln(color.Output, line)
+			if log.IsDebug() || show {
+				fmt.Println(line)
```

Comment:
shouldn't this write to `out`, instead of directly to stdout?
Why do we have `out` in this function?

Reply:
Bit confused, but how it works is that: If show is passed in or debug, it'll print this directly.

We have `cmdOut` in the function purely for when we *dont* use --show or debug. Since if build fails, it'll output the entire build output to the user.

Does that make sense?

Reply:
this function has `our io.Writer` argument. If I'm not mistaken this argument is there  represent `os.Stdout`, you should use that.

Reply:
what I mean is to use `fmt.Fprintf(out,`

Reply:
@kadel Been updated! :+1: